website/docs/configuration/provider-requirements.html.md

```diff
@@ -0,0 +1,299 @@
+with the type name "terraform".
+
+There is also an existing provider with the source address
+`hashicorp/terraform`, which is an older version of the now-built-in provider
+that was used by older versions of Terraform. `hashicorp/terraform` is not
+compatible with Terraform v0.11 or later and should never be declared in a
+`required_providers` block.
+
+### In-house Providers
+
+Some organizations develop their own providers to allow interacting with
+proprietary systems, and wish to use these providers from Terraform without
+publishing them on the public Terraform Registry.
+
+One option for distributing such a provider is to run an in-house _private_
+registry, by implementing
+[the provider registry protocol](/docs/internals/provider-registry-protocol.html).
+
+Running an additional service just to distribute a single provider internally
+may be undesirable though, so Terraform also supports
+[other provider installation methods](https://github.com/hashicorp/terraform/blob/master/website/docs/commands/cli-config.html.markdown#provider-installation),
+including placing provider plugins directly in specific directories in the
+local filesystem, via _filesystem mirrors_.
+
+All providers must have a [source address](#source-addresses) that includes
+(or implies) the hostname of a host registry, but for an in-house provider that
+you intend only to distribute from a local filesystem directory you can choose
+an artificial hostname in a domain your organization controls and use that to
+mark your in-house providers.
+
+For example, if your corporate domain were `example.com` then you might choose
+to use `terraform.example.com` as your artificial hostname, even if that
+hostname doesn't actually resolve in DNS. You can then choose any namespace and
+type you wish to represent your in-house provider under that hostname, giving
+a source address like `terraform.example.com/examplecorp/ourcloud`:
+
+```hcl
+terraform {
+  required_providers {
+    mycloud = {
+      source  = "terraform.example.com/examplecorp/ourcloud"
+      version = ">= 1.0"
+    }
+  }
+}
+```
+
+To make version 1.0.0 of this provider available for installation from the
+local filesystem, choose one of the
+[implied local mirror directories](/docs/commands/cli-config.html#implied-local-mirror-directories)
+and create a directory structure under it like this:
+
+```
+terraform.example.com/examplecorp/ourcloud/1.0.0
+```
+
+Under that `1.0.0` directory, create one additional directory representing the
+platform where you are running Terraform, such as `linux_amd64` for Linux on
+an AMD64/x64 processor, and then place the provider plugin executable and any
+other needed files in that directory.
+
+The provider plugin executable file might therefore be at the following path,
+on a Windows system for the sake of example:
+
+```
+terraform.example.com/examplecorp/ourcloud/1.0.0/windows_amd64/terraform-provider-ourcloud.exe
```

Comment:
For a windows example, should the slashes be `\`?

Reply:
Forward slashes work on Windows too and so we've tended to just use `/` exclusively in our examples for compatibility, though I admit it's a bit odd in this case because it's specifically an example _for_ Windows.

With that said, it's really only a Windows example because it had to be an example of _something_. Given that Windows is the outlier here, I think I'd rather change it to a different OS than confuse everyone else by using backslashes. :thinking: